frame/lottery/src/lib.rs

```diff
@@ -444,7 +444,7 @@ impl<T: Config> Module<T> {
 	// You should call this function with different seed values until the random
 	// number lies within `u32::MAX - u32::MAX % n`.
 	fn generate_random_number(seed: u32) -> u32 {
-		let random_seed = T::Randomness::random(&(T::ModuleId::get(), seed).encode());
+		let (random_seed, _) = T::Randomness::random(&(T::ModuleId::get(), seed).encode());
```

Comment:
same here, probably should add a TODO to patch this, or we should patch it in this PR.

Reply:
I am thinking out loud here: The logic could technically bloat a bit due to this, in the worse case I presume. I am not familiar with this pallet for example but in case you really need a fresh random number in `on_initialize`, and if it is not ready, then you need to queue to work and re-try later, which might have a whole array of side effects.